03week/mastermind.js

```diff
@@ -20,43 +83,83 @@ function printBoard() {
 function generateSolution() {
   for (let i = 0; i < 4; i++) {
     const randomIndex = getRandomInt(0, letters.length);
-    solution += letters[randomIndex];
+    generatedCode += letters[randomIndex];
   }
 }
 
 function getRandomInt(min, max) {
   return Math.floor(Math.random() * (max - min)) + min;
 }
 
-function generateHint() {
-  // your code here
+function mastermind(guess) {
+  if (guess === generatedCode) {
+    console.log('You win!');
+    return;
+  } else {
+    console.log('Keep trying');
+  }
+  generateHint(guess);
 }
 
-function mastermind(guess) {
-  solution = 'abcd'; // Comment this out to generate a random solution
-  // your code here
+
+function generateHint(guess) {
+  let gueArray = guess.split('');
+  let genArray = generatedCode.split('');
+  let counter=0;
+
+  for(let i = 0; i < gueArray.length;i++)
+  {
+    for(let j=0; j < genArray.length;j++)
+    {
+      if (gueArray[i]===genArray[j])
+        counter+=1;
+    }
+  }
+  console.log('you have '+counter+' correct letters');
+
+  let positionCounter=0;
+  if (gueArray[0] === genArray[0]) {
```

Comment:
Consolidate lines 121-132 using a for loop and `indexof()` to check your guess against solution.